tensorbay/cli/utility.py

```diff
@@ -26,95 +26,213 @@
 from .tbrn import TBRN
 
 _Callable = TypeVar("_Callable", bound=Callable[..., None])
+INDENT = " " * 4
 
 
-class ContextInfo(NamedTuple):
+class ContextInfo:
     """This class contains command context."""
 
-    access_key: str
-    url: str
-    profile_name: str
-    config_parser: ConfigParser
+    def __init__(self, access_key: str, url: str, profile_name: str):
+        self.access_key = access_key
+        self.url = url
+        self.profile_name = profile_name
+        self.config_parser = self._read_config()
 
+    @staticmethod
+    def _get_config_filepath() -> str:
+        """Get the path of the config file.
 
-def _implement_cli(
-    ctx: click.Context, access_key: str, url: str, profile_name: str, debug: bool
-) -> None:
-    config_parser = update_config(read_config())
-    ctx.obj = ContextInfo(access_key, url, profile_name, config_parser)
+        Returns:
+            The path of the config file.
 
-    if debug:
-        logging.basicConfig(level=logging.DEBUG)
-    else:
-        logger.disabled = True
+        """
+        home = "USERPROFILE" if is_win() else "HOME"
+        return os.path.join(os.environ[home], ".gasconfig")
 
+    @staticmethod
+    def _is_updated(profile_section: SectionProxy) -> bool:
+        return "\n" in profile_section.get("accesskey", "\n")
 
-def _get_config_filepath() -> str:
-    """Get the path of the config file.
+    @staticmethod
+    def _parse_profile_info(profile_info: str) -> Tuple[str, str]:
+        """Parse the profile information to get accessKey and URL.
 
-    Returns:
-        The path of the config file.
+        Arguments:
+            profile_info: The profile information read from the config file.
 
-    """
-    home = "USERPROFILE" if is_win() else "HOME"
-    return os.path.join(os.environ[home], ".gasconfig")
+        Returns:
+            A tuple containing the accessKey and the url of profile_name read from the config file.
 
+        """
+        values = profile_info.split("\n")
+        if len(values) == 2:
+            return values[1], ""
+        return values[1], values[2]
 
-def is_win() -> bool:
-    """Determine whether the current system is Windows.
+    def _read_config(self, config_filepath: Optional[str] = None) -> ConfigParser:
```

Comment:
`_read_config` is only called once, can we just expand it?
also the input argument `config_filepath` is never used? can we remove it?